File: pueblo-golf-tournament-api/Dto/Incoming/AuthenticateLoginDto.cs
using System.ComponentModel.DataAnnotations;

namespace pueblo_golf_tournament_api.Dto.Incoming
{
    public class AuthenticateLoginDto
    {
        [Required]
        public string Username { get; set; } = "";
        [Required]
        public string Password { get; set; } = "";
    }
}

File: pueblo_golf_tournament_mobile/lib/api/requests/authentication/authenticate-login-dto.dart
import 'package:json_annotation/json_annotation.dart';

part 'authenticate-login-dto.g.dart';

@JsonSerializable()
class AuthenticateLoginDto {
  final String username;
  final String password;

  AuthenticateLoginDto({required this.username, required this.password});

  factory AuthenticateLoginDto.fromJson(Map<String, dynamic> json) =>
      _$AuthenticateLoginDtoFromJson(json);

  Map<String, dynamic> toJson() => _$AuthenticateLoginDtoToJson(this);
}
